Automattic/wp-calypso
# Simple Payments: gutenberg block UI

Adds basic UI for Simple payments Gutenberg block.

## Screenshots

#### UI:
<img src="https://user-images.githubusercontent.com/87168/47922789-903ffd80-dec0-11e8-9801-a05b8d01067b.png" width="500" />

#### UI screaming warnings:
<img src="https://user-images.githubusercontent.com/87168/47922790-903ffd80-dec0-11e8-97ae-9d0fc70fd018.png" width="500" />

#### Preview in editor:

<img src="https://user-images.githubusercontent.com/87168/47882842-760c0e00-de33-11e8-8f30-e407d25e42bb.png" width="250"/>

#### End result at the site:

<img src="https://user-images.githubusercontent.com/87168/47883031-342f9780-de34-11e8-8060-740f5f678670.png" width="250"/>


## Todo / known issues
- [ ] Merge API changes https://github.com/Automattic/jetpack/pull/10402
- [ ] Land D19907-code
- [x] Missing input error states
   - [x] Improve or add value validation in the block and in the API
   - [x] Highlighting invalid inputs after block de-select
   - [ ] Show errors coming from the API
- [x] Small styling adjustments to other elements:
    - [x] Email help text font size
    - [ ] horizontal alignment of currency and price inputs
- [x] Link "PayPal" word in email help text to `paypal.com`
- [x] PayPal image in preview mode doesn't work; need to figure out how to ship png files
- [ ] Limit prices to `>= 1.00`
- [x] _when I unfocus the block, my 0.5 USD becomes $0.00_ (this is probably because of the backend that sanitizes prices to integers right now)
- [x] Cleanup `view.scss` — now it's just copied over from Jetpack plugin and we need very little from it for the preview
- [ ] Block is always enabled regardless of the site plan (Poseidon)

## Testing instructions

### Test in Jetpack
* Jurassic Ninja: https://jurassic.ninja/create?jetpack-beta&branch=update/simple-payments-rest-api-meta&gutenpack&gutenberg&calypsobranch=add/gutenpack-simplepayments-ui
* Connect Jetpack, purchase "premium" or "business" plan
* Create new post in Gutenberg and add "Payment button" block
* Open the post and see the button
* Open the post in Calypso classic editor and see the button
* Open the post in Calypso using this branch and Gutenberg and see the button: https://calypso.live/gutenberg/post?branch=add/gutenpack-simplepayments-ui

### Test in Calypso
- Apply patch to your sandbox: D19907-code
- Open https://calypso.live/gutenberg/post?branch=add/gutenpack-simplepayments-ui
- Choose a site with Simple payments enabled (purchase "premium" or "business" plan)

### Manual checking
Apply these patches and cross-test the block in different editors:
- This PR
- https://github.com/Automattic/jetpack/pull/10402
- D19907-code
